python/job_runner/runner.py

```diff
@@ -0,0 +1,87 @@
+import json
+import os
+
+from job_runner.job import Job
+from job_runner.reporting.message import Init, Finish
+
+
+class JobRunner(object):
+
+    def __init__(self, jobs_file="jobs.json"):
+        try:
+            with open(jobs_file, "r") as json_file:
+                self.jobs_data = json.load(json_file)
+        except ValueError as e:
+            raise IOError(
+                "Job Runner failed to load JSON-file.{}".format(str(e)))
+
+        os.umask(int(self.jobs_data["umask"], 8))
+
+        self._data_root = self.jobs_data.get("DATA_ROOT")
+        if self._data_root:
+            os.environ["DATA_ROOT"] = self._data_root
+
+        self.simulation_id = self.jobs_data.get("run_id")
+        self.ert_pid = self.jobs_data.get("ert_pid")
+        self.global_environment = self.jobs_data.get("global_environment")
+        self.global_update_path = self.jobs_data.get("global_update_path")
+        self.job_data_list = self.jobs_data["jobList"]
+
+        if self.simulation_id is not None:
+            os.environ["ERT_RUN_ID"] = self.simulation_id
+
+        self.jobs = []
+        for index, job_data in enumerate(self.job_data_list):
+            self.jobs.append(Job(job_data, index))
+
+        self._set_environment()
+        self._update_path()
+
+    def run(self, names_of_jobs_to_run):
+        # if names_of_jobs_to_run, create job_queue which contains jobs that
+        # are to be run.
+        if not names_of_jobs_to_run:
+            job_queue = self.jobs
+        else:
+            job_queue = [j for j in self.jobs if j.name()
+                         in names_of_jobs_to_run]
+
+        init_message = Init(job_queue, self.simulation_id, self.ert_pid)
+
+        unused = set(names_of_jobs_to_run) - set([j.name() for j in job_queue])
+        if unused:
+            init_message.with_error(
+                "{} does not exist. Available jobs: {}"
+                .format(unused, [j.name() for j in self.jobs])
+            )
+            yield init_message
+            raise StopIteration()
+        else:
+            yield init_message
+
+        all_jobs_successful = True
+        for job in job_queue:
```

Comment:
Job runner should terminate if one job is not successful

Reply:
There should be a test for this ;)

Reply:
`job_dispatch` kills itself with `SIGKILL` if some job fails, but this should be documented in `JobRunner`, since that's the place you are likely to look.

We should also test this specifically in the `job_dispatch` testing.

Reply:
Taken care of by job_dispatch.py

Reply:
Yes, let's have `job_dispatch` kill on unsuccessful `Finish`, and have file reporter report everything it needs to on `Exited`.